docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,77 @@ The key corresponding parts of the implementation are as follows.
+
+- A value of `EraParams` for `F` and for `G` (ie `Shape [F, G]`), which decomposes as follows.
+    - The duration of every slot in the epoch as a [`NominalDiffTime`](https://hackage.haskell.org/package/time-1.12.2/docs/Data-Time-Clock.html#t:NominalDiffTime).
+    - The duration of every epoch in the era as a number of slots.
+    - The era's _safe zone_, which depends on the `singleEraTransition` function; see below.
+        - If the era's safe zone is `d`, then `singleEraTransition` cannot return `Just e` for the ledger state resulting from some block `x` (possibly the "genesis block") (possibly from an earlier era, and not even necessarily the previous era) unless there are at least `d` slots between `x` and `e`, ie `slot(x) + d < firstSlot(e)`.
+        - The fundamental motivation for the HFC combinator is to enable a chain to eventualy seemlessly transition into eras that do not yet exist.
+          Therefore, the sequence `[F, G]` may currently be known to be incomplete; ie there will be a `[F, G, H]` in the future, but `H` doesn't _yet_ exist.
+          For such `[F, G]` chains, `G`'s `EraParams` should still specify a finite safe zone, since the blocks created by nodes running `[F, G]` will eventually be validated by future nodes running `[F, G, H]` (eg see PR <https://github.com/input-output-hk/ouroboros-network/pull/3754>).
+        - On the other hand, if `G` is instead currently known to be the _final_ era for some `[F, G]` chain then the `EraParams` used for `G` on that chain can instead specify a so-called _indefinite_ safe zone.
+          If an era has an indefinite safe zone, then `singleEraTransition` must return `Nothing` (TODO we could implement this as an override, should we? ... Is that already effectively the case?)
+          Using an indefinite safe zone prematurely (eg by accident) may cause confusion, but it's not necessarily fatal; minor exceptional cases in the ledger rules etc might suffice (eg see PR <https://github.com/input-output-hk/cardano-ledger/pull/2785>).
+- An instance of `CanHardFork [F, G]`, which decomposes as follows.
+    - A function `LedgerConfig F -> LedgerConfig G -> LedgerState F -> Bound {- starting slot&epoch&time-since-genesis of G -} -> SlotNo -> Except OutsideForecastRange (LedgerView G)`.
+      This function will be used for forecasting the ledger view across the era transition; see "How does cross-era forecasting work?" above.
+    - A function `ConsensusConfig F -> ConsensusConfig G -> EpochNo {- start of G -} -> ChainDepState F -> ChainDepState G`.
+      This function will be applied to the chain-dependent state immediately after the last `F` header.
+      Then the protocol rules for `G` will be used to tick that state to the slot of the first `G` header.
+      This use of `G`'s ticking rules instead of `F`'s to cross the end of the last epoch of `F` is subtle and somewhat counterintuitive (eg see Issue <https://github.com/input-output-hk/cardano-ledger/issues/3491>).
```

Comment:
Maybe remove the link to that issue here, as it only affects ledger states and is repeated just below? Instead, we could write that we haven't yet encountered any problems with the translate-then-tick scheme for `ChainDepState`s (except maybe a incorrect (as in: not-as-specified) handling of non-neutral extra entropy at the TPraos->Praos depending on how you read the ledger specs).

Reply:
I think maybe I pasted the wrong link here. Let me check.

Reply:
I added a new sentence that mentions extra entropy, for example.